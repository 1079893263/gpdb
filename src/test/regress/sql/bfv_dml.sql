-- MPP-21622 Update with primary key: only sort if the primary key is updated
--
-- Aside from testing that bug, this also tests EXPLAIN of an DMLActionExpr
-- that ORCA generates for plans that update the primary key.
create table update_pk_test (a int primary key, b int) distributed by (a);
insert into update_pk_test values(1,1);

explain update update_pk_test set b = 5;
update update_pk_test set b = 5;
select * from update_pk_test order by 1,2;

explain update update_pk_test set a = 5;
update update_pk_test set a = 5;
select * from update_pk_test order by 1,2;
