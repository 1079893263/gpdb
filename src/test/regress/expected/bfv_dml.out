-- MPP-21622 Update with primary key: only sort if the primary key is updated
--
-- Aside from testing that bug, this also tests EXPLAIN of an DMLActionExpr
-- that ORCA generates for plans that update the primary key.
create table update_pk_test (a int primary key, b int) distributed by (a);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "update_pk_test_pkey" for table "update_pk_test"
insert into update_pk_test values(1,1);
explain update update_pk_test set b = 5;
                             QUERY PLAN                              
---------------------------------------------------------------------
 Update (slice0; segments: 3)  (rows=1 width=14)
   ->  Seq Scan on update_pk_test  (cost=0.00..1.01 rows=1 width=14)
 Settings:  optimizer=off
 Optimizer status: legacy query optimizer
(4 rows)

update update_pk_test set b = 5;
select * from update_pk_test order by 1,2;
 a | b 
---+---
 1 | 5
(1 row)

explain update update_pk_test set a = 5;
ERROR:  Cannot parallelize an UPDATE statement that updates the distribution columns
update update_pk_test set a = 5;
ERROR:  Cannot parallelize an UPDATE statement that updates the distribution columns
select * from update_pk_test order by 1,2;
 a | b 
---+---
 1 | 5
(1 row)

